geps/gep-1651.md

```diff
@@ -0,0 +1,154 @@
+# GEP-1651: GatewayAddress Scope
+
+* Issue: [#1651](https://github.com/kubernetes-sigs/gateway-api/issues/1651)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## TLDR
+
+Allow users to configure a GatewayAddress so that it is only accessible within
+a specific scope (ie. public/cluster local)
+
+## Goals
+
+- Cluster local gateways can be created in a declarative way `ClusterLocal` and
+  are only accessible within a single cluster
```

Comment:
How can this contract be assured? Typically a Service of type `ClusterIP` is only reachable within a cluster, but we cannot assume that it's __only__ reachable within the cluster.

Reply:
> How can this contract be assured? Typically a Service of type ClusterIP is only reachable within a cluster, but we cannot assume that it's only reachable within the cluster.

Do we know how K8s does this now? I don't see too many options because even if the test assumes the runner is off cluster - someone might run the conformance tests _in_ a cluster.

Reply:
Perhaps the naming should be `ClusterIP` or `Cluster`.  I think this doc is assuming a default Kubernetes cluster setup where the Pod and Service IPs are only accessible via the overlay network set up by the cluster's CNI.  In the case where one or both IP ranges are routed outside the cluster CNI's scope (e.g. [this section of Calico's documentation](https://docs.tigera.io/calico/latest/networking/determine-best-networking#pod-ip-routability-outside-of-the-cluster)), then the name should reflect that the IP reachability matches that of the cluster.

The request here that we've heard from users of Knative is that they'd like to be able to use the same IP reachability controls for "internal" services that they use for other Kubernetes resources.


In terms of testing, I can imagine that we could validate a `ClusterIP` or `ServiceIP`-scoped Gateway address is within the cluster's IP range (defaulting to `10/8`, but probably with a flag to adjust).

In any case, I think there's also value in allowing other scopes like "VPC", but those are likely to be cloud-provider (really, IP-routing-provider) dependent.

Reply:
I changed this the constant to `cluster` and defined it as

> The address is only reachability within the bounds defined by the [cluster's network model](https://kubernetes.io/docs/concepts/cluster-administration/networking/#how-to-implement-the-kubernetes-network-model)